### ISSUE TITLE
[data grid] localization of date field from system, not from theme

### ISSUE THREAD

**Issue by NickGlackikh** [mui/mui-x#10553]
### Duplicates

- [X] I have searched the existing issues

### Current behavior 😯

pagination section (works fine)

![image](https://github.com/mui/material-ui/assets/44984714/49e8dcb0-be2a-4c51-a433-330c7497169a)

calendar section when row is being updated (month name and buttons are in russian)

![image](https://github.com/mui/material-ui/assets/44984714/cfb1991d-21d1-45dc-abe1-d1c2a2edab28)

### Context 🔦

Hi! I have a problem with MUI grid community.
I have connected the ukrainian localization for my page and everything is OK except datepicker for date field. It keeps using system localization. 

here is my code
import { ukUA } from '@mui/x-data-grid';
import { ukUA as pickersukUA } from '@mui/x-date-pickers/locales';
import { ukUA as coreukUA } from '@mui/material/locale';
import { createTheme, ThemeProvider } from '@mui/material/styles';

**const theme** = createTheme(
  {
    palette: {
      primary: { main: '#1976d2' },
    },
  },
  ukUA, // x-data-grid translations
  pickersukUA, // x-date-pickers translations
  coreukUA // core translations
);
........
 **<ThemeProvider theme={theme}>**
    <div className='mainGridWrapper'>
        <DataGrid 
        editMode='cell'
        
        onRowClick={props.handleRowClick}
        
        processRowUpdate={(updatedRow) =>
          {
          props.onRowUpdate(updatedRow);
          return updatedRow;}
        }

columns={[{ field: 'id'},
        { field:'edit', headerName:'', 
        headerClassName: 'super-app-theme--header',
        flex:5,
        align:'center',
        sortable: false,
        renderCell:(strVal)=>{return(
          <Tooltip title="Редагувати">
          <IconButton onClick={()=>{console.log(strVal.row.datap)}}>
             <EditOutlinedIcon />
          </IconButton>
          </Tooltip>
        )}},
        

                 **{ field: 'datap',** 
                  type: 'date',

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**:

**Comment by Lev-Shapiro** [comment]
Hi, I'll be happy to help you with this problem!

But initially there are two problems:

**Firstly,** the code you've provided won't reproduce the problem, because it has some problems including not using calendar field and not using the theme that was initialized, but not called. 

**Secondly, ** I want to understand if I understood your issue correctly - you have successfully translated the pagination section to Ukrainian, however you weren't able to translate the date picker, so it was left with the initial language of your app ( Russian ) and what you want is to translate it to Ukrainian too, correct?

**Comment by Lev-Shapiro** [comment]
By the way I've made a basic actually working example, the least thing to do is to add the theme

```
<DataGrid
        editMode="cell"
        processRowUpdate={(updatedRow) => updatedRow}
        columns={[
          { field: "id", type: "number" },
          {
            field: "date",
            headerName: "Date",
            type: "date",
            editable: true,
          },
        ]}
        rows={[
          {
            id: 1,
            date: new Date(),
          },
          {
            id: 2,
            date: new Date(),
          },
          {
            id: 3,
            date: new Date(),
          },
          {
            id: 4,
            date: new Date(),
          },
        ]}
      />
```

**Comment by NickGlackikh** [comment]
> btw I've made a basic actually working example:
> 
> Theme & imports init:
> 
> ```
> import { ukUA as coreukUA } from "@mui/material/locale";
> import { ThemeProvider, createTheme } from "@mui/material/styles";
> import { DataGrid, ukUA } from "@mui/x-data-grid";
> import { ukUA as pickersukUA } from "@mui/x-date-pickers/locales";
> 
> const theme = createTheme(
>   {
>     palette: {
>       primary: { main: "#1976d2" },
>     },
>   },
>   ukUA, // x-data-grid translations
>   pickersukUA, // x-date-pickers translations
>   coreukUA // core translations
> );
> ```
> 
> JSX:
> 
> ```
>       <ThemeProvider theme={theme}>
>         <DataGrid
>           editMode="cell"
>           processRowUpdate={(updatedRow) => updatedRow}
>           columns={[
>             { field: "id", type: "number" },
>             {
>               field: "date",
>               headerName: "Date",
>               type: "date",
>               editable: true,
>             },
>           ]}
>           rows={[
>             {
>               id: 1,
>               date: new Date(),
>             },
>             {
>               id: 2,
>               date: new Date(),
>             },
>             {
>               id: 3,
>               date: new Date(),
>             },
>             {
>               id: 4,
>               date: new Date(),
>             },
>           ]}
>         />
>  </ThemeProvider>
> ```

Yes, You`ve understood my problem right.. Maybe MUI grid doesn`t have such properties.
The problem is definitely in standart mui grid calendar. When I try to add MUI DatePicker to renderEditCell, the localization works fine. (But I got another errors :))

**Comment by Lev-Shapiro** [comment]
Could you tell what's the problem you get if using `DatePicker` inside `renderEditCell`? What I get is TimePicker but without  any buttons, you have the same problem or am I doing something wrong simply?

**Comment by alexfauquette** [comment]
@NickGlackikh As mentioned [here](https://mui.com/x/react-data-grid/recipes-editing/#usage-with-mui-x-date-pickers) the data grid uses the native date pickers (which localization are based on the system)

If you want to localize the calendar with `@mui/x-date-pickers` you need to do some modification as shown in the demo

**Comment by electrocnic** [comment]
> [@NickGlackikh](https://github.com/NickGlackikh) As mentioned [here](https://mui.com/x/react-data-grid/recipes-editing/#usage-with-mui-x-date-pickers) the data grid uses the native date pickers (which localization are based on the system)

Your link mentions "#usage-with-mui-x-date-pickers" but there is no chapter named like this on that page anymore.

> If you want to localize the calendar with `@mui/x-date-pickers` you need to do some modification as shown in the demo
I could not find a demo that instructs me how to localize the date format within a MUI-X DataGrid's Filter-Menu for a Date-typed column.

I initially tried the approach with 
```
<LocalizationProvider dateAdapter={AdapterDateFns} adapterLocale={de}>
  {children}
</LocalizationProvider>;
```
and then had to realize that this sadly does not apply to the DataGrid at all.
It should imo.

Then I found this issue, it is still open, so let me ask of any updates on this?
